### PR TITLE
!!! [v6r12] Dirac compile web

### PR DESCRIPTION
**Pull request by zmathe**

### CODE REVIEW

File: FrameworkSystem/Service/SystemAdministratorHandler.py [hunk 1]
```diff
@@ -225,12 +225,30 @@ def export_updateSoftware( self, version, rootPath = "", gridVersion = "" ):
...
+    #TODO: It can be removed when we do not use anymore the old portal...
     if webFlag:
       extensionList.append( 'Web' )
     if extensionList:
       cmdList += ['-e', ','.join( extensionList )]
-
+    
+    webPortal = gConfig.getValue( '/LocalInstallation/WebApp', False )  # this is the new portal
+    if webPortal:  # we have a to compile the new web portal...
+      extensionList.append( 'Web' )
```
Comment (atsareg) [inline FrameworkSystem/Service/SystemAdministratorHandler.py]: Should it be WebApp extension ?

**Comment by atsareg** [comment]
I suspect that this fix is not correct. You try to compile the WebApp code before updating its software. This will compile the previously installed code only

**Comment by zmathe** [comment]
Yes, You are right as the cmdList will be executed after when the compile. So, the compile has to be moved after the cmdList is executed.

**Comment by atsareg** [comment]
Note also that after the dirac-install, the dirac-webapp-compile command will be in the path. No need to execute it with python explicitely. Also, this fix should rather go to the dirac-postInstall.py script to be placed in the top directory of this package. See, DIRACWeb as an example. SysAdmin service should not bother about these details

**Comment by zmathe** [comment]
Ah I see if the dirac-webapp-compile is in the path then the compilation will be easier ...
This can not go to the dirac-postInstall.py. This is used to install extjs...

**Comment by atsareg** [comment]
I do not understand why it can not go to the dirac-postInstall.py. Can you elaborate please ?

**Comment by zmathe** [comment]
The dirac-postInstall.py always executed when we instal DIRAC:
python dirac-install -t server $installCfg
source $installDir/bashrc
dirac-configure $installCfg $DEBUG
dirac-setup-site $DEBUG

This time we have to install extjs as we know extjs does not exists as well we have to install SenchaCMD used to compile the code. But this is the preparation step.
This script also do not executed when you update DIRAC.
As well we do not want to re-install every time extjs and sencha.

**Comment by atsareg** [comment]
I think we are speaking about different things. I mean dirac-postInstall script inside the WebAppDIRAC package which is executed after the package is (re)installed during the overall software update. This script can check if the code compilation is necessary or not before doing it.

**Pull request merged**